File: app/src/main/java/com/example/lucas/haushaltsmanager/Utils/BundleUtils.java
package com.example.lucas.haushaltsmanager.Utils;

import android.os.Bundle;

public class BundleUtils {
    private final Bundle mBundle;

    public BundleUtils(Bundle bundle) {
        mBundle = bundle == null ? new Bundle() : bundle;
    }

    public String getString(String key, String defaultValue) {

        return mBundle.containsKey(key) ? mBundle.getString(key) : defaultValue;
    }

    public long getLong(String key, long defaultValue) {

        return mBundle.containsKey(key) ? mBundle.getLong(key) : defaultValue;
    }

    public Object getParcelable(String key, Object defaultValue) {

        return mBundle.containsKey(key) ? mBundle.getParcelable(key) : defaultValue;
    }

    public boolean hasMapping(String key) {
        return mBundle.containsKey(key);
    }
}

File: app/src/main/java/com/example/lucas/haushaltsmanager/Utils/BundleUtils.kt
package com.example.lucas.haushaltsmanager.Utils

import android.os.Bundle

class BundleUtils(val bundle: Bundle) {
    fun findString(key: String): String {
        if (bundle.containsKey(key)) {
            return bundle.getString(key)!!
        }

        throw IllegalArgumentException("Bundle contains no key with value of '$key'!")
    }

    fun getString(key: String, defaultValue: String): String {
        return bundle.getString(key, defaultValue);
    }

    fun getLong(key: String, defaultValue: Long): Long {
        return bundle.getLong(key, defaultValue)
    }

    fun getParcelable(key: String, defaultValue: Any?): Any? {
        if (bundle.containsKey(key)) {
            return bundle.getParcelable(key)!!
        }

        return defaultValue
    }

    fun hasMapping(key: String): Boolean {
        return bundle.containsKey(key)
    }
}